### PR TITLE
requeue_scrub when kick_object_context_blocked

### PR DESCRIPTION
**Pull request by Abhishekvrshny**
http://tracker.ceph.com/issues/13035

### CODE REVIEW

**Comment by dachary** [comment]
<pre>
$ commit=c7f430a ; picked_from=$(git show --no-patch --pretty=%b $commit  | perl -ne 'print if(s/.*cherry picked from commit (\w+).*/$1/)') ; diff -u --ignore-matching-lines '^[^+-]'   <(git show $picked_from) <(git show $commit)
--- /dev/fd/63	2015-09-15 15:56:52.762575739 +0200
+++ /dev/fd/62	2015-09-15 15:56:52.762575739 +0200
@@ -21,7 +25,7 @@
  void ReplicatedPG::kick_object_context_blocked(ObjectContextRef obc)
  {
    const hobject_t& soid = obc->obs.oi.soid;
--  map<hobject_t, list<OpRequestRef>, hobject_t::BitwiseComparator>::iterator p = waiting_for_blocked_object.find(soid);
+-  map<hobject_t, list<OpRequestRef> >::iterator p = waiting_for_blocked_object.find(soid);
 -  if (p == waiting_for_blocked_object.end())
 -    return;
 -
</pre>

File: src/osd/ReplicatedPG.cc [hunk 1]
```diff
@@ -8088,19 +8088,18 @@ void ReplicatedPG::add_object_context_to_pg_stat(ObjectContextRef obc, pg_stat_t
...
   if (obc->is_blocked()) {
     dout(10) << __func__ << " " << soid << " still blocked" << dendl;
     return;
   }
 
-  list<OpRequestRef>& ls = p->second;
-  dout(10) << __func__ << " " << soid << " requeuing " << ls.size() << " requests" << dendl;
-  requeue_ops(ls);
-  waiting_for_blocked_object.erase(p);
+  map<hobject_t, list<OpRequestRef>, hobject_t::BitwiseComparator>::iterator p = waiting_for_blocked_object.find(soid);
```
Comment (dachary) [inline src/osd/ReplicatedPG.cc]: s/, hobject_t::BitwiseComparator//

**Comment by dachary** [comment]
The logic of this backport does not seem to require https://github.com/ceph/ceph/pull/5470. It should work just by removing **hobject_t::BitwiseComparator** as suggested. When you do, make sure you update the conflict message to reflect this decision. See http://tracker.ceph.com/projects/ceph-releases/wiki/HOWTO_backport_commits around **if there is a conflict**.

**Comment by dachary** [comment]
@liewegas does this backport look good to merge ? It passed a run of the hammer rados suite ( see http://tracker.ceph.com/issues/13356#note-11 and  for details ).

**Comment by liewegas** [comment]
:+1:

**Pull request merged**